### PR TITLE
KYLIN-3715 ProjectL2Cache don't be invalidated when adding override config in cu…

### PR DESCRIPTION
**Pull request by walhl**
…be level

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

File: core-cube/src/main/java/org/apache/kylin/cube/CubeManager.java [hunk 1]
```diff
@@ -151,11 +151,13 @@ protected CubeInstance initEntityAfterReload(CubeInstance cube, String resourceN
 
         @Override
         public void onProjectSchemaChange(Broadcaster broadcaster, String project) throws IOException {
-            for (IRealization real : ProjectManager.getInstance(config).listAllRealizations(project)) {
+            ProjectManager projectManager = ProjectManager.getInstance(config);
+            for (IRealization real : projectManager.listAllRealizations(project)) {
```
Comment (shaofengshi) [inline core-cube/src/main/java/org/apache/kylin/cube/CubeManager.java]: Hello, I can't see the difference; can you elaborate on it? Thank you.

File: core-cube/src/main/java/org/apache/kylin/cube/CubeManager.java [hunk 2]
```diff
@@ -151,11 +151,13 @@ protected CubeInstance initEntityAfterReload(CubeInstance cube, String resourceN
...
                 if (real instanceof CubeInstance) {
                     reloadCubeQuietly(real.getName());
                 }
             }
+            projectManager.reloadProjectL2Cache(project);
```
Comment (walhl) [inline core-cube/src/main/java/org/apache/kylin/cube/CubeManager.java]: “onProjectSchemaChange ” method updated all cubeinstance under the given project,  but cubeinstances in ProjectL2Cache  are not updated.  I  added reloading "ProjectL2Cache " after updating cubeinstance.

Comment (shaofengshi) [reply core-cube/src/main/java/org/apache/kylin/cube/CubeManager.java]: Sorry, I missed this line;
In Broadcaster.java line 253, it clears project L2 cache before calling "onProjectSchemaChange", so does it need to reload the project L2 cache again?

Comment (walhl) [reply core-cube/src/main/java/org/apache/kylin/cube/CubeManager.java]: during calling "onProjectSchemaChange", 
there will invoke CubeDescSyncListener.onProjectSchemaChange method. During this method,  "ProjectManager.getInstance(config).listAllRealizations(project) " will lead to loading previous cubeinstance in ProjectL2Cache .  then there will invoke CubeManager.onProjectSchemaChange()  method.

Comment (walhl) [reply core-cube/src/main/java/org/apache/kylin/cube/CubeManager.java]: During invoking "CubeManager.onProjectSchemaChange()" method,  projectL2Cache  did't updated.

Comment (shaofengshi) [reply core-cube/src/main/java/org/apache/kylin/cube/CubeManager.java]: I see, thank you Walhl; If moving "projectManager.reloadProjectL2Cache(project);" to Broadcaster line 256, after all listeners are notified, will that be better?

**Pull request merged**